src/util.jl

```diff
@@ -79,6 +79,57 @@ function hilbert{T<:Real}(x::AbstractArray{T})
     out
 end
 
+# Evaluate a window function at n points, returning both the window
+# (or nothing if no window) and the squared L2 norm of the window
+compute_window(::Nothing, n::Int) = (nothing, n)
+function compute_window(window::Function, n::Int)
+    win = window(n)::Vector{Float64}
+    norm2 = sumabs2(win)
+    (win, norm2)
+end
+function compute_window(window::AbstractVector, n::Int)
+    length(window) == n || error("length of window must match input")
+    (window, sumabs2(window))
+end
+
+backward_plan{T<:Union(Float32, Float64)}(X::AbstractArray{Complex{T}}, Y::AbstractArray{T}) =
+    FFTW.Plan(X, Y, 1, FFTW.ESTIMATE, FFTW.NO_TIMELIMIT).plan
+
+function istft{T<:Union(Float32, Float64)}(S::AbstractMatrix{Complex{T}}, wlen::Int, overlap::Int; nfft=nextfastfft(wlen), window::Union(Function,AbstractVector,Nothing)=nothing)
+    winc = wlen-overlap
+    win, norm2 = compute_window(window, wlen)
+    if win != nothing 
+      win² = win.^2
+    end
+    nframes = size(S,2)-1
+    outlen = nfft + nframes*winc
+    out = zeros(T, outlen)
+    tmp1 = similar(S[:,1])
```

Comment:
This would should be `Array(eltype(S), size(S, 1))` to avoid making an unnecessary copy and to ensure that the array is actually an Array (which is not necessarily the case if `S` is an AbstractMatrix).